Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -201,10 +201,10 @@ class DatabaseContainer: NSPersistentContainer {
     func write(_ actions: @escaping (DatabaseSession) throws -> Void, completion: @escaping (Error?) -> Void) {
         writableContext.perform {
             log.debug("Starting a database session.", subsystems: .database)
-            // Clear the FetchCache
-            FetchCache.clear()
             do {
+                FetchCache.clear()
```

Comment:
Better have one before the do, and another one after the catch ends. That way you avoid repetition

Reply:
we need to clear the cache before calling `save` since `save` kicks off any observers. `didChange` callbacks sometimes use DB and this causes the cache to have old info, and it causes CoreData errors

Reply:
Good point. What about a local block so that we don't need to repeat `FetchCache.clear()` so often then?

Reply:
i couldn't figure it out, how would that look?